src/radix_trie.c

```diff
@@ -563,6 +565,12 @@ enum status radix_trie_free (struct radix_trie *r) {
 
     /* perform deep free of radix_trie starting at the root */
     radix_trie_deep_free(r->root);
+
```

Comment:
why not move the loop to free the flags up before the radix_trie_deep_free and then you can simply change that call to radix_trie_free (it does the deep free and free of the structure). Then you can remove the rt_free below the the flag loop.

Reply:
I might be missing something here? This is the function radix_trie_free that is being modified so we can't remove the rt_free at the bottom. Also it may be confusing for the function to recurse.

Reply:
i see now, you are right. just ignore that last part of the comment.